templates/azure-cluster-identity/asomanagedazurecluster-identity-ref.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+kind: AzureASOManagedControlPlane
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  identityRef:
```

Comment:
This obviously didn't work w/ the new ASO API the way it does w/ the existing AKS API:

```
Running kubectl apply --kubeconfig /root/.kube/config -f -
stderr:
Error from server (BadRequest): error when creating "STDIN": AzureASOManagedControlPlane in version "v1alpha1" cannot be handled as a AzureASOManagedControlPlane: strict decoding error: unknown field "spec.identityRef"
```

@nojnhuh Is there add'l work need to enable workload-identity parity w/ the existing AKS API, or am I just doing it wrong?

Reply:
You shouldn't need to make any changes for the ASOAPI types. They don't use AzureClusterIdentity at all, just the native ASO auth: https://azure.github.io/azure-service-operator/guide/authentication/credential-format/. The e2e test already uses Workload ID.

Reply:
ACK. It was failing earlier to provision so I'll undo changes and figure out what's wrong.